ci/scripts/c_glib_build.sh

```diff
@@ -28,17 +28,38 @@ build_root=${2}
 : ${BUILD_DOCS_C_GLIB:=OFF}
 with_doc=$([ "${BUILD_DOCS_C_GLIB}" == "ON" ] && echo "true" || echo "false")
 
-export PKG_CONFIG_PATH=${ARROW_HOME}/lib/pkgconfig
+
+if [ -n "${MSYSTEM:-}" ]; then
+    # Fix ARROW_HOME when running under MSYS2
+    export ARROW_HOME="$(cygpath --unix "${ARROW_HOME}")"
+fi
+
+meson_pkg_config_path="${ARROW_HOME}/lib/pkgconfig:${ARROW_HOME}/bin/pkgconfig"
```

Comment:
Do we need `:${ARROW_HOME}/bin/pkgconfig`?

Reply:
Yes, when I test the build in Windows the pkgconfig files are installed to the bin directory, and without adding that I get an error:
```
Run-time dependency arrow found: NO (tried pkgconfig and cmake)

c_glib\meson.build:108:10: ERROR: Dependency "arrow" not found, tried pkgconfig and cmake
```

Reply:
If we remove https://github.com/apache/arrow/pull/41599/files#r1602805729 , we can remove this path too.

Reply:
Yes you're right, I copied that from the C++ build (`.github/workflows/cpp.yml`) but had forgotten about it sorry. I'm not sure why the C++ build is configured like that. I've removed setting the LIBDIR variable and the extra pkgconfig path now.